### ISSUE TITLE
[Custom Packages] Moving to Input packages

### ISSUE THREAD

**Issue by jamiehynds** [elastic/integrations#8435]
Our 'custom packages' currently state that changing the dataset will send the data to a different index. However, these packages create a template such as `logs-udp.generic` matching `logs-udp.generic-*` indices. This contradicts the text saying that changing the dataset will send the data to a different index.

Currently, the data itself will go into the new destination, however the initial component template is made to match the incorrect pattern.

In order to resolve this, our custom packages need to be convert to input packages (i.e. `type: input`)

We need to test the user experience when upgrading from a current custom package to the new input package. Are there breaking changes involved and to what extend? Depending on the experience, we may need to discuss deprecation of the current packages and shipping new v2 input packages (but hopefully that can be avoided). Fleet team will need to be engaged to discuss. 

List of packages to convert:

- [ ] CEL
- [ ] httpjson (consider depreaction here to avoid confusion over CEL vs httpjon)??
- [ ] AWS (S3 and Cloudwatch inputs)
- [ ] Azure Blob Storage
- [ ] Google Cloud Storage
- [ ] Google Pub/Sub
- [ ] HTTP Endpoint
- [ ] Journald
- [ ] Kafka
- [ ] Log
- [ ] TCP
- [ ] UDP
- [ ] Windows

**Comment by andrewkroh** [comment]
> We need to test the user experience when upgrading from a current custom package to the new input package.

Agree. The most important thing is that the upgrade requires no user intervention and that the Fleet stream ID does not change so that input state is not lost. Please do a test for this using one of the stateful inputs and verify that the existing state continues to be used after the upgrade.

**Comment by andrewkroh** [comment]
The blog post at https://www.elastic.co/blog/customize-data-ingestion-elastic-input-packages can be edited to include the converted packages when they are done. @ishleenk17 can probably aid in that edit if we let her know.

**Comment by ishleenk17** [comment]
> The blog post at https://www.elastic.co/blog/customize-data-ingestion-elastic-input-packages can be edited to include the converted packages when they are done. @ishleenk17 can probably aid in that edit if we let her know.

Sure, let me know when these are migrated. We can add it to the blog.

**Comment by taylor-swanson** [comment]
@ishleenk17, [per this comment](https://github.com/elastic/integrations/issues/8435#issuecomment-1945437359), we have a new package to add to the list, `Custom UDP Logs (udp)`.
